osu.Game.Rulesets.Catch/Objects/Drawables/IHasCatchObjectState.cs

```diff
@@ -12,7 +12,9 @@ namespace osu.Game.Rulesets.Catch.Objects.Drawables
     /// </summary>
     public interface IHasCatchObjectState
     {
-        PalpableCatchHitObject HitObject { get; }
+        CatchObjectType ObjectType { get; }
+
+        double StartTime { get; }
```

Comment:
Immediately I'm not sure I agree with this premise of having that data added to all catch DHOs and copied to the caught object model. Why isn't this just the hitobject itself? Hitobjects (the models) are not pooled after all. Can't the dropped objects just take a reference to the HOs?

The main reason that I'm opposed to this is that catch object types are encoded already, in the hitobject types themselves. Adding this `ObjectType` enum to boot seems like unnecessary duplication - this sort of stuff should really have a singular source of ground truth.

I'm not sure if it's worth to review the rest of this since this is a pretty central piece of this entire change, so abstaining with the rest for now.

Reply:
The main motivation for removing `HitObject` reference was that its properties could change after the object is caught in the editor.
It turns out, however, in the current state of PR, `HitObject` properties being changed after caught doesn't cause issues. [The way](https://github.com/ppy/osu/blob/fa94703b9b663e443896a3971bfc2c2102350b93/osu.Game.Rulesets.Catch/Skinning/Default/CatchHitObjectPiece.cs#L51)`StartTime` currently used doesn't matter `StartTime` as long as it is after the current time. And the type of `HitObject` cannot be changed.
I'll make it references to a `HitObject` for now to make the PR simpler.

Reply:
A bit hard to me to respond to that without knowing exactly in which circumstances the properties could change. Let's just cross that bridge when we get to it, but I'd probably honestly rather prefer cloning hitobjects to this myself.